tsl/src/compression/create.c

```diff
@@ -1064,21 +1080,27 @@ tsl_process_compress_table_add_column(Hypertable *ht, ColumnDef *orig_def)
 	int32 orig_htid = ht->fd.id;
 	char *colname = orig_def->colname;
 	TypeName *orig_typname = orig_def->typeName;
+	bool new_column = false; /* need to know if the column already exists so we don't try to add it
+								to the catalog again */
+	bool no_compression_table = false;
 
 	coloid = LookupTypeNameOid(NULL, orig_typname, false);
 	compresscolinfo_init_singlecolumn(&compress_cols, colname, coloid);
+
 	if (TS_HYPERTABLE_HAS_COMPRESSION_TABLE(ht))
 	{
 		int32 compress_htid = ht->fd.compressed_hypertable_id;
 		Hypertable *compress_ht = ts_hypertable_get_by_id(compress_htid);
-		add_column_to_compression_table(compress_ht, &compress_cols);
+		new_column = add_column_to_compression_table(compress_ht, &compress_cols);
 	}
 	else
 	{
+		no_compression_table = true;
 		Assert(TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht));
 	}
 	/* add catalog entries for the new column for the hypertable */
-	compresscolinfo_add_catalog_entries(&compress_cols, orig_htid);
+	if (new_column || no_compression_table)
```

Comment:
add catalog entries only if this is a new column, otherwise you will have duplicates.

```suggestion
	if (new_column )
```